|Original version of this patch by Ralph Sennhauser <sera@gentoo.org>
|Updated version for 3.5.25 by tomboy64 <tomboy64@sina.cn>
|
|Added in 3.5.4, make false default for Gentoo
|
|* Add a new wrapper.java.command.resolve property to control whether or not the
|  Wrapper tries to resolve any symbolic links in the Java command, specified
|  with the wrapper.java.command property.  Historically, it has always done so,
|  but some jvm started applications like run-java-tool on Gentoo will fail if
|  it is run directly as they have a check to make sure it is launched via a
|  symbolic link.
|

--- src/c/wrapper.c.old	2014-07-21 09:12:55.000000000 +0200
+++ src/c/wrapper.c	2014-07-21 09:13:22.000000000 +0200
@@ -4908,7 +4908,7 @@
     if (!path) {
         log_printf(WRAPPER_SOURCE_WRAPPER, LEVEL_WARN, TEXT("The configured wrapper.java.command could not be found, attempting to launch anyway: %s"), *para);
     } else {
-        replacePath = getBooleanProperty(properties, TEXT("wrapper.java.command.resolve"), TRUE);
+        replacePath = getBooleanProperty(properties, TEXT("wrapper.java.command.resolve"), FALSE);
         if (replacePath == TRUE) {
             free(*para);
             *para = malloc((_tcslen(path) + 1) * sizeof(TCHAR));

